qchem/pennylane_qchem/qchem/structure.py

```diff
@@ -1059,6 +1061,163 @@ def excitations_to_wires(singles, doubles, wires=None):
     return singles_wires, doubles_wires
 
 
+def force_constants(
+    H,
+    x,
+    idx,
+    hessian,
+    ansatz,
+    params,
+    dev,
+    interface="autograd",
+    diff_method="best",
+    optimize="False",
+    delta=0.01,
+):  # pylint: disable=too-many-arguments
```

Comment:
This is a heavy function :-(. It might reveal why having driver-like functions is not very appealing. Instead, the lego philosophy is preferable. The exception here could be that computing the second derivative of the energy might not be straightforward for the user. I am open to discuss this point.

Reply:
Perhaps this makes more sense as a transform function? This way, the user passes their own qnodes etc., avoiding needing to try and take into account all interfaces yourself here

Reply:
Agree with both of you: this is too heavy and very different from the style of code we are striving for with pennylane. @agran2018 Let's have a chat to discuss how this could be turned into a transform and figure out if it's worth implementing